### PR TITLE
Mainly simplification of some passages and fixing some issues reported by static code analysis

### PR DESCRIPTION
**Pull request by floscher**
I wanted to try out the integrations with [CodeCov](https://codecov.io/github/floscher/josm-mapillary-plugin) and [Codacy](https://www.codacy.com/app/floscher/josm-mapillary-plugin), so these changes are now in a pull request instead of being directly committed.

As I wrote in the heading, I tried to simplify some overly complicated passages of the code and fixed some issues reported by static code analysis, focused on readability and performance.

### CODE REVIEW

File: src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java [hunk 1]
```diff
@@ -16,11 +16,15 @@
  */
 public class PluginState {
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java]: Codacy Issue found: [A class which only has private constructors should be final](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1585067294/issues/source?bid=3023888&fileBranchId=3024747#l17)

Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java]: Codacy Issue found: [A class which only has private constructors should be final](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1585158142/issues/source?bid=3023888&fileBranchId=3024747#l17)

**Comment by nokutu** [comment]
The webpages look pretty cool. What are they actually used for? Error detection?
I suppose they both are completely unrelated, do they actually overlap each other?

**Comment by floscher** [comment]
@nokutu CodeCov checks the code coverage (basically the percentage of code that is tested, currently about 22%) and would display as one failed test at the bottom of the pull request if the coverage would decrease when merging the pull request.
Codacy is indeed for error detection and ensures that some style guidelines are followed. For each pull request and commit it shows how many issues are introduced and fixed. This would also show as a failed check at the bottom of the pull request, if the code quality decreased.
They are completely separate tools for different purposes. Codacy could check coverage too, but I like the coverage interface of CodeCov better, because it's more detailed. And they have a cool browser plugin, that displays code coverage directly on GitHub.

File: src/org/openstreetmap/josm/plugins/mapillary/oauth/UploadUtils.java [hunk 2]
```diff
@@ -59,7 +59,7 @@
     private final List<MapillaryAbstractImage> images;
     private final UUID uuid;
     private final boolean delete;
-    ThreadPoolExecutor ex;
+    private ThreadPoolExecutor ex;
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/oauth/UploadUtils.java]: Codacy Issue found: [Private field 'ex' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1585492936/issues/source?bid=3023888&fileBranchId=3024747#l62)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java (hunk 1 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java]: Codacy Issue found: [A class which only has private constructors should be final](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1585492945/issues/source?bid=3023888&fileBranchId=3024747#l17)

File: src/org/openstreetmap/josm/plugins/mapillary/oauth/UploadUtils.java (hunk 2 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/oauth/UploadUtils.java]: @nokutu and these line notes are automatically added where new issues are introduced. I'm more and more convinced of these tools. I think in the future, I'll more often open pull requests before pushing to SVN to have my changes reviewed before they are published to production.

File: src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java [hunk 3]
```diff
@@ -14,13 +14,17 @@
...
  *
  */
-public class PluginState {
+public final class PluginState {
 
-  private static int runningDownloads = 0;
+  private static int runningDownloads;
   /** Images that have to be uploaded. */
-  protected static int imagesToUpload = 0;
+  protected static int imagesToUpload;
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java]: Codacy Issue found: [Avoid protected fields in a final class.  Change to private or package access.](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1585734601/issues/source?bid=3023888&fileBranchId=3024747#l21)

**Comment by floscher** [comment]
I've now committed the changes to SVN, so all six commits are now in `master`.

File: src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java (hunk 3 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/PluginState.java]: Codacy Issue found: [Avoid protected fields in a final class.  Change to private or package access.](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1585734601/issues/source?bid=3023888&fileBranchId=3024747#l21)

**Pull request closed without merging**